rpcserver.go

```diff
@@ -1512,6 +1512,44 @@ func (r *rpcServer) ListChannels(ctx context.Context,
 			linkActive = link.EligibleToForward()
 		}
 
+		// Next, we'll determine whether we should add this channel to
+		// our list depending on the type of channels requested to us.
+		isActive := peerOnline && linkActive
+		isPublic := dbChannel.ChannelFlags&lnwire.FFAnnounceChannel ==
+			lnwire.FFAnnounceChannel
+
+		switch {
+		case in.Active && in.Inactive || !in.Active && !in.Inactive:
```

Comment:
I would recommend just making the "in params" `active_only`, `inactive_only` etc, then just filter out channels that don't match the criteria (and you could return error early if two incpompatible flags are both set). Should simplify this code significantly!